core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -558,15 +561,34 @@ private void collectCatalogItems(String sourceYaml, Map<?,?> itemMetadata, List<
             throw new IllegalStateException("Could not resolve plan once id and itemType are known (recursive reference?): "+sourceYaml);
         }
         String sourcePlanYaml = planInterpreter.getPlanYaml();
-        
-        CatalogItemDtoAbstract<?, ?> dto = createItemBuilder(itemType, symbolicName, version)
+
+        CatalogItemBuilder itemBuilder = createItemBuilder(itemType, symbolicName, version)
             .libraries(libraryBundles)
             .displayName(displayName)
             .description(description)
             .deprecated(catalogDeprecated)
             .iconUrl(catalogIconUrl)
-            .plan(sourcePlanYaml)
-            .build();
+            .plan(sourcePlanYaml);
+        if (itemType.equals(CatalogItemType.ENTITY) || itemType.equals(CatalogItemType.TEMPLATE)) {
+            String stringServiceType = getFirstAs(itemAsMap, String.class, "type").orNull();
+            // This is not a single item but a services based YAML
+            if (stringServiceType == null) {
+                final List<Map<?, ?>> services = getFirstAs(itemAsMap, List.class, "services").orNull();
+                if (services.isEmpty()) {
+                    throw new IllegalStateException("Services are empty: " + sourceYaml);
+                }
+                stringServiceType = services.get(0).get("type").toString();
+            }
+            Class<?> serviceType = JavaBrooklynClassLoadingContext.create(mgmt).tryLoadClass(stringServiceType).orNull();
+            if (serviceType == null) {
+                final CatalogItem<?, ?> relatedCatalogItem = getCatalogItem(stringServiceType, "");
+                if (relatedCatalogItem != null) {
+                    serviceType = JavaBrooklynClassLoadingContext.create(mgmt).tryLoadClass(relatedCatalogItem.getJavaType()).orNull();
+                }
+            }
+            itemBuilder.tag(BrooklynTags.newInterfacesTag(ClassUtils.getAllInterfaces(serviceType)));
```

Comment:
[The code seems to be the same](https://git-wip-us.apache.org/repos/asf?p=commons-lang.git;a=blob;f=src/main/java/org/apache/commons/lang3/ClassUtils.java;h=0a7480ff8d991e555be679f86377dc19d3eb64d7;hb=HEAD#l433) as `Reflections.getAllInterfaces(Class<?>)`. Although, the `Reflections` class says in its headers:
>  Reflection utilities ("borrowed" from cloudsoft monterey).

So I would still use `ClassUtils`.